src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -48,44 +46,61 @@ public PlaceGuideServlet(String userId) {
   public static final String DESCRIPTION_INPUT = "description";
   public static final String LENGTH_INPUT = "length";
   public static final String IMAGE_KEY_INPUT = "imageKey";
-  public static final String PLACE_GUIDE_TYPE_PARAMETER = "placeGuideType";
+  public static final String PLACE_GUIDE_QUERY_TYPE_PARAMETER = "placeGuideType";
+  public static final String REGION_CORNERS_PARAMETER = "regionCorners";
 
-  private final PlaceGuideRepository placeGuideRepository = 
+  private final PlaceGuideRepository placeGuideRepository =
       PlaceGuideRepositoryFactory.getPlaceGuideRepository(RepositoryType.DATASTORE);
 
-  private enum PlaceGuideQueryType {
-    ALL, 
-    CREATED_ALL, 
-    CREATED_PUBLIC, 
-    CREATED_PRIVATE 
-  }
-
-  /**
-   * Saves the recently submitted place guide data.
-   */
+  /** Saves the recently submitted place guide data. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     PlaceGuide placeGuide = getPlaceGuideFromRequest(request);
     placeGuideRepository.savePlaceGuide(placeGuide);
     response.sendRedirect("/createPlaceGuide.html");
   }
 
-  /**
-   * Returns the data of the placeguide(s) asked by the user who is currently logged in.
-   */
+  /** Returns the data of the placeguide(s) asked by the user who is currently logged in. */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String placeGuideType = request.getParameter(PLACE_GUIDE_TYPE_PARAMETER);
-    PlaceGuideQueryType queryType = PlaceGuideQueryType.valueOf(placeGuideType);
-    List<PlaceGuide> placeGuides = getPlaceGuides(queryType);
+    String placeGuideQueryTypeString = request.getParameter(PLACE_GUIDE_QUERY_TYPE_PARAMETER);
+    PlaceGuideQueryType placeGuideQueryType =
+        PlaceGuideQueryType.valueOf(placeGuideQueryTypeString);
+    GeoPt northEastCorner = null;
+    GeoPt southWestCorner = null;
+    if (placeGuideQueryType.requiresCoordinates()) {
```

Comment:
when do we not require coordinates?

Reply:
We don't require coordinates for the original queries(ALL_PUBLIC, CREATED_PUBLIC, ...). I would not remove them completely until we make sure that the new approach that we discussed today will work.